docs/CI-CD.md

```diff
@@ -0,0 +1,93 @@
+  action triggered on the latest tag is that it gets the benefit of a full (from scratch) build
+  as opposed to using pre-cached artifacts. This is done to ensure that EVE builds can be rebuilt
+  in the future in a self-contained manner but it does, of course, increase the build time dramatically.
+* If Publish gets triggered on a release tag that is NOT considered to be the `latest` (for example when
+  building .z patch releases for older x.y.(z-1) releases) the OCI registry artifacts only get a matching
+  `x.y.z` tag and no additional aliases are generated. The build is also typically much faster since it
+  uses pre-cached artifacts in OCI registry if the corresponding source package in the EVE tree didn't
+  change. Note, however, that building a patch release for the previous latest tag becomes a new latest
+  tag and thus subject to the latest tag treatment (e.g. `x.y.z` being latest means `x.y.(z+1)` will now
+  compare as greater than any other tag since it is greater than `x.y.z` itself).
+* If Publish gets triggered on a `master` branch all EVE packages in an OCI registry get two tags assigned:
+    1. symbolic tag `snapshot` (to facilitate `docker pull lfedge/eve-pillar:snapshot` type of usage)
+    2. source-derived unique tag
+  Source-derived unique tag is expected to be a SHA-like ID that changes only when any part of the
+  source tree corresponding to the EVE packages changes and thus allows for lazy (CAS-like) caching
+  of build updates. For all packages with actual source code this is linuxkit generate git tree sha
+  (e.g. `lfedge/eve-pillar:b8c0b1813a7819ee9eefdc77f1883da5613e7017`) for rootfs and its package this
+  is a synthetic version ID generated by the makefile of the form `0.0.0-master-<git sha>`.
+* If Publish gets triggered on any other branch (although this is rare and is not supported by default
+  in our Publish action) the Makefile logic will be identical to what's happening with a master branch
+  with the only two exceptions being around the name of the symbolic tag. Instead of a `snapshot` it
+  will be either:
+    1. `x.y.z-<# commits on top>-<git sha>` (e.g. `5.15.1-6-g6658d671f`) for branches that can be
+        traced back to a git tag
+    2. `branch name` (e.g. `foo`) for branches that do NOT contain a single git tag in their history
+
+## Pulls
+
+Pulls is, effectively, a much more constrained version of the Publish action. The reason it needs
+to be constrained is that it triggers on any pull request submitted to the master branch and that
+may include anonymous pull requests containing malicious code in them. Builds triggered by this
+action do NOT deposit artifacts into the regular OCI registry. The registry that is being used
+is a quarantined one over at `evebuild/danger` and the only artifact that is ever being deposited
+there is a rootfs container tagged as `pr-ID` (e.g. `docker pull evebuild/danger:pr-12345` will
+fetch a rootfs container for Pull Request 12345).
```

Comment:
This still requires credentials, even if constrained. Why should we push any image at all to a registry, even just to the quarantined area, on PRs from untrusted sources? What do we need them for?

Reply:
The way it is done doesn't require credentials in the VM that touches untrusted code. I suppose I can go into details of how it is implemented -- is this what you're suggesting @deitch ?

Reply:
I was more curious about the "why". Why would we push out built images from a PR?